src/internal.c

```diff
@@ -26474,14 +26512,23 @@ int SendCertificateVerify(WOLFSSL* ssl)
             goto exit_scv;
     }
     else
+#endif
+    if (ssl->fragOffset != 0) {
+        /* We should be in the sending state. */
+        if (ssl->options.asyncState != TLS_ASYNC_END) {
+            ret = BAD_STATE_E;
+            goto exit_scv;
+        }
+    }
+    else
 #endif
     {
         /* Reset state */
         ret = 0;
         ssl->options.asyncState = TLS_ASYNC_BEGIN;
         XMEMSET(args, 0, sizeof(ScvArgs));
-    #ifdef WOLFSSL_ASYNC_CRYPT
-        ssl->async.freeArgs = FreeScvArgs;
+    #ifdef WOLFSSL_ASYNC_IO
+        ssl->async->freeArgs = FreeScvArgs;
```

Comment:
Also here, I think in the cleanup of the function:
```    /* Final cleanup */
    FreeScvArgs(ssl, args);
    FreeKeyExchange(ssl);
```
FreeScvArgs is invoked twice.

Reply:
Fixed.